### PR TITLE
Move Reports to the lib

### PR DESCRIPTION
**Pull request by samoht**
This will need a few more tests to check that the output is correct.

### CODE REVIEW

**Comment by samoht** [comment]
I've now added simple tests (more complex ones are failing, I'll fix them in separate PRs)

**Comment by samoht** [comment]
This is ready to merge - I have plans to improve the intermediate format (to avoid all this nasty insertion of indents) but I'll do this in separate PRs.

File: test/pp/test.ml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+(*
+ * Copyright (c) 2021 Thomas Gazgnaire <thomas@gazagnaire.org>
```
Comment (MagnusS) [inline test/pp/test.ml:2]:
```suggestion
 * Copyright (c) 2021 Thomas Gazagnaire <thomas@gazagnaire.org>
```

**Review by MagnusS — APPROVED** [review]

File: test/pp/test.ml (hunk 1 above)
Comment (samoht) [reply test/pp/test.ml]: 🤣

**Pull request merged**